pkg/controllers/replica-controller/handler.go

```diff
@@ -257,7 +261,22 @@ func (c *CStorVolumeReplicaController) cVREventHandler(
 		if isCVRCreateStatus(cvrObj) {
 			return c.cVRAddEventHandler(cvrObj, fullVolName)
 		}
-		return c.getCVRStatus(cvrObj)
+
+		if cvrObj.Annotations[restoreCompletedAnnotation] == "true" {
+			targetIp, err := volumereplica.GetTargetIp(fullVolName)
+			if err != nil {
+				return "", err
+			}
+
+			if targetIp != cvrObj.Spec.TargetIP {
+				err := volumereplica.SetTargetIp(fullVolName, cvrObj.Spec.TargetIP)
+				if err != nil {
+					return "", err
+				}
+			}
+
+			delete(cvrObj.Annotations, restoreCompletedAnnotation)
```

Comment:
Can we also delete `isRestoreVol` annotation? Removing `IsRestoreVol` annotations will help in ephemeral cases(Cases where the pool is recreated then VolumeReplicas also will be recreated). In this case when VolumeReplica/zvol is recreated then targetIP will be set by VolumeReplica controller only if `IsRestoreVol` annotation doesn't exist.

Reply:
sure